validator/sawtooth_validator/execution/scheduler_serial.py

```diff
@@ -49,6 +49,9 @@ def __init__(self, squash_handler, first_state_hash):
         self._last_in_batch = []
         self._last_state_hash = first_state_hash
 
+    def __del__(self):
```

Comment:
All components that hold scarce resources should clean them up automatically when they go away. When in python do you worry about an a file handle not being closed correctly. See the reference to __del__ here: https://docs.python.org/3/reference/datamodel.html. The CandidateBlock calls either finalize if the block is build or cancel if the block is discarded via the CandidateBlock.__del__ method.